upgrade/upgrade-scripts/upgrade-container

```diff
@@ -42,14 +42,17 @@ function create_cleanup() {
 	destroy "$CONTAINER"
 }
 
-function get_mounted_rootfs_name() {
-	local rootfs
+function mount_upgrade_container_dataset() {
+	[[ -n "$1" ]] || die "failed to provide dataset to mount"
+	[[ -n "$2" ]] || die "failed to provide directory for mount"
 
-	rootfs=$(zfs list -Ho name /)
-	[[ -n "$rootfs" ]] ||
-		die "failed to determine the mounted root filesystem name"
+	mkdir -p "$2" || die "failed to create directory '$2'"
```

Comment:
Not related to this review, but since we are running the upgrade scripts from the app-stack, could they have the same issue as we see here: https://jira.delphix.com/browse/LX-1852

Reply:
I haven't seen any problems with the delphix CLI after an upgrade.

Reply:
When you run the not-in-place upgrade test, can you double-check that the permissions for `/export` are indeed right? If so, then either we don't use the same umask in the app-stack on Linux, the umask somehow doesn't get propagated to the script, or the `/export` directory's permissions are modified somewhere else.

Reply:
I have run "not-in-place" upgrade multiple times before using the scripts, and have successfully used the delphix CLI afterwards without issue.

I can do this again, and can also do it via the delphix GUI.. but either way, I think this is unrelated to this change.

Reply:
After a not-in-place upgrade via the GUI:
```
$ ls -alhd /export/ /export/home/ /export/home/cli/
drwxr-xr-x 3 root root  3 Apr 24 20:24 /export/
drwxr-xr-x 4 root root  4 Apr 23 22:51 /export/home/
drwxr-xr-x 3 cli  staff 7 Apr 23 22:51 /export/home/cli/
```
I'm able to use the CLI too:
```
± ssh admin@ps-upgrade.dlpxdc.co
Password: 
ip-10-110-208-210> ls
Children
about
action
...
```

Reply:
You can do the same yourself with the following:
```
$ download-latest-image internal-dev
$ sudo unpack-image -x internal-dev.upgrade.tar.gz
```
Then go into the GUI and run the upgrade as normal.

Reply:
That's interesting, thanks for testing this. I'll need to see how come this is different from the migration case.